tests/test_glue.py

```diff
@@ -587,3 +588,50 @@ def test_remove_statements_validation_error(self):
                 "actions": [{"type": "remove-statements", "statement_ids": "matched"}],
             }
         )
+
+    def test_catalog_cloudtrail_event(self):
+        session_factory = self.replay_flight_data("test_catalog_cloudtrail_event")
+        session = session_factory()
+        client = session.client("glue")
+        before_cat_setting = client.get_data_catalog_encryption_settings()
+        self.assertEqual(before_cat_setting.get('DataCatalogEncryptionSettings').get(
+            'EncryptionAtRest').get('CatalogEncryptionMode'), 'DISABLED')
+        self.assertEqual(before_cat_setting.get('DataCatalogEncryptionSettings').get(
+            'EncryptionAtRest').get('SseAwsKmsKeyId'), None)
+        p = self.load_policy(
+            {
+                "name": "net-change-rbp-cross-account",
+                "resource": "glue-catalog",
+                "mode": {
+                    "type": "cloudtrail",
+                    "role": "arn:aws:iam::644160558196:role/CloudCustodianRole",
+                    "events": [
+                        {
+                            "source": "glue.amazonaws.com",
+                            "event": "PutResourcePolicy",
+                            "ids": "awsRegion"
+                        }
+                    ],
+                },
+                "filters": [
+                    {"type": "cross-account"}
+                ],
+                "actions": [
+                    {
+                        "type": "set-encryption",
```

Comment:
Just realized I did something weird here on filter/action mismatch. Its an action discrepancy between internal/external running c7n version. Will update in a few.

Reply:
resolved in 59f5d71